decoders/sent/pd.py

```diff
@@ -0,0 +1,805 @@
+
+        # compute CRC
+        crc_bits = ''
+
+        for (b2, b3) in zip(self.bits2[6:], self.bits3[6:]):
+            crc_bits += b2
+            crc_bits += b3
+
+        computed_crc = EnhancedSerialCrc.INIT
+
+        for i in range(4):
+            x = int(crc_bits[i*6:(i+1)*6], 2)
+            computed_crc = EnhancedSerialCrc.update(computed_crc, x)
+
+        computed_crc = EnhancedSerialCrc.finalize(computed_crc)
+
+        if computed_crc != crc:
+            self.decoder.put_error_at(self.frame_starts[0], self.frame_starts[7], f'Serial: CRC mismatch: 0x{computed_crc:02X}')
+
+        # clean up
+        self._reset()
+
+        # pretend we just read the last part of this message
+        self.bits3 = '0'
+        self.bits2 = '0'
+        self.frame_starts = [self.decoder.pulse_falling1]
+
+    def on_frame_start(self):
+        self.frame_starts.append(self.decoder.pulse_falling0)
+
+    def on_frame_end(self, error):
+        if None is error or not error:
+            if len(self.bits3) == SlowProtocolEnhanced.FRAMES_FOR_MESSAGE + 1 and self.bits3.startswith('01111110'):
+                self._on_message_complete()
+        else:
+             self._reset()
+
+    def update(self, status_nibble):
+        self.bits3 += str(int((status_nibble & SlowProtocolEnhanced.BIT3_FLAG) == SlowProtocolEnhanced.BIT3_FLAG))
+        self.bits2 += str(int((status_nibble & SlowProtocolEnhanced.BIT2_FLAG) == SlowProtocolEnhanced.BIT2_FLAG))
+
+        if len(self.bits3) > SlowProtocolEnhanced.FRAMES_FOR_MESSAGE + 1:
+            self.bits3 = self.bits3[1:]
+            self.bits2 = self.bits2[1:]
+            self.frame_starts = self.frame_starts[1:]
+
+
+class SamplerateError(Exception):
+    pass
+
+
+class ClockTickTimeError(Exception):
+    pass
+
+
+class SlowFormatError(Exception):
+    pass
+
+
+class Sent:
+    FAST_FORMAT_TO_FRAME_LENGTH = {
+        'H.1': 6,
+        'H.2': 3,
+        'H.3': 4,
+        'H.4': 6,
+        'H.5': 6,
+        'H.6': 6,
+        'H.7': 6,
+    }
+
+    class ProtocolState:
+        NONE = 'NONE'
+        CALIBRATION = 'CALIBRATION'
+        STATUS = 'STATUS'
+        DATA = 'DATA'
+        CRC = 'CRC'
+        PAUSE = 'PAUSE'
+
+    PULSE_PAUSE_MODE_MAYBE = 'maybe'
+    PULSE_PAUSE_MODE_YES = 'yes'
+    PULSE_PAUSE_MODE_NO = 'no'
+    PULSE_PAUSE_MODES = (PULSE_PAUSE_MODE_MAYBE, PULSE_PAUSE_MODE_YES, PULSE_PAUSE_MODE_NO)
+
+    SLOW_CHANNEL_FORMAT_NONE = 'None'
+    SLOW_CHANNEL_FORMAT_SHORT = 'Short'
+    SLOW_CHANNEL_FORMAT_ENHANCED_C0 = 'Enhanced C0'
+    SLOW_CHANNEL_FORMAT_ENHANCED_C1 = 'Enhanced C1'
+    SLOW_CHANNEL_FORMATS = (SLOW_CHANNEL_FORMAT_NONE, SLOW_CHANNEL_FORMAT_SHORT, SLOW_CHANNEL_FORMAT_ENHANCED_C0, SLOW_CHANNEL_FORMAT_ENHANCED_C1)
+
+class Decoder(srd.Decoder):
+    api_version = 3
+    id = 'sent'
+    name = 'SENT'
+    longname = 'Single Edge Nibble Transmission'
+    desc = 'One-wire serial bus.'
+    license = 'mit'
```

Comment:
Ok. I think MIT should be ok , based on some other PDs also licensed under MIT (ade77xx, enc28j60, nrf905....)